source/slang/core.meta.slang

```diff
@@ -1490,27 +1484,85 @@ for (auto op : unaryOps)
         if (op.flags & BOOL_RESULT) resultType = "bool";
 
         // scalar version
-        sb << fixity;
         sb << "__generic<T : " << op.interface << ">\n";
-        sb << "__intrinsic_op(" << int(op.opCode) << ") " << resultType << " operator" << op.opName << "(" << qual << "T value);\n";
+        sb << "__prefix __intrinsic_op(" << int(op.opCode) << ") " << resultType << " operator" << op.opName << "(" << "T value);\n";
 
         // vector version
         sb << "__generic<T : " << op.interface << ", let N : int> ";
-        sb << fixity;
-        sb << "__intrinsic_op(" << int(op.opCode) << ") vector<" << resultType << ",N> operator" << op.opName << "(" << qual << "vector<T,N> value);\n";
+        sb << "__prefix __intrinsic_op(" << int(op.opCode) << ") vector<" << resultType << ",N> operator" << op.opName << "(vector<T,N> value);\n";
 
         // matrix version
         sb << "__generic<T : " << op.interface << ", let N : int, let M : int> ";
-        sb << fixity;
-        sb << "__intrinsic_op(" << int(op.opCode) << ") matrix<" << resultType << ",N,M> operator" << op.opName << "(" << qual << "matrix<T,N,M> value);\n";
+        sb << "__prefix __intrinsic_op(" << int(op.opCode) << ") matrix<" << resultType << ",N,M> operator" << op.opName << "(matrix<T,N,M> value);\n";
     }
 }
 
-for (auto op : binaryOps)
+}}}}
+
+__generic<T : __BuiltinArithmeticType>
+[__unsafeForceInlineEarly]
```

Comment:
It's a small point but I'm now wondering if 'unsafe' seems to be gaining weight from it's meaning in rust, but that is not what is meant here.

Reply:
Yes, we aren't using `unsafe` in a way that is consistent with the Haskell/Rust/Swift precedent, and we should maybe think about that.

For now I'm not too worried because this isn't a feature we ever plan to expose to users with its current name.